lib/Sema/TreeTransform.h

```diff
@@ -5228,11 +5230,47 @@ bool TreeTransform<Derived>::TransformFunctionTypeParams(
   return false;
 }
 
+template<typename Derived>
+bool TreeTransform<Derived>::TransformBoundsAnnotations(
+  BoundsAnnotations &Annot, bool &Changed) {
+  BoundsExpr *ExistingBounds = Annot.getBoundsExpr();
+  BoundsExpr *NewBounds = ExistingBounds;
+  if (ExistingBounds) {
+    ExprResult Result = getDerived().TransformExpr(ExistingBounds);
+    if (Result.isInvalid())
+      return true;
+    NewBounds = dyn_cast<BoundsExpr>(Result.get());
+    if (!NewBounds) {
+      assert("unexpected dynamic cast failure");
+      return true;
+    }
+  }
+
+  InteropTypeExpr *ExistingIType = Annot.getInteropTypeExpr();
+  InteropTypeExpr *NewIType = ExistingIType;
+  if (ExistingIType) {
+    ExprResult Result = getDerived().TransformExpr(ExistingIType);
+    if (Result.isInvalid())
+      return true;
+    NewIType = dyn_cast<InteropTypeExpr>(Result.get());
+    if (!NewIType) {
+      assert("unexpected dynamic cast failure");
```

Comment:
Same here.

Reply:
There's an isa<VarDecl> test in the enclosing test, so we don't need to check for null here.